engine/src/io/data_parser/CSVParser.cpp

```diff
@@ -115,22 +115,12 @@ std::unique_ptr<ral::frame::BlazingTable> csv_parser::parse(
 			return column_indices[i1] < column_indices[i2];
 		});
 
-		// TODO columns_out should change (gdf_column_cpp)
-		for(size_t i = 0; i < csv_table.tbl->num_columns(); i++) {
-			if(csv_table.tbl->get_column(i).type().id() == cudf::type_id::STRING) {
-//				NVStrings * strs = static_cast<NVStrings *>(table_out.get_column(i)->data);
-//				NVCategory * category = NVCategory::create_from_strings(*strs);
-//				std::string column_name(table_out.get_column(i)->col_name);
-//				columns_out[idx[i]].create_gdf_column(category, table_out.get_column(i)->size, column_name);
-//				gdf_column_free(table_out.get_column(i));
-			} else {
-				//TODO create_gdf_column anymore
-				//columns_out[i].create_gdf_column(csv_table.tbl->get_column(i), csv_table.metadata.column_names[i]);
-			}
-		}
-		return std::make_unique<ral::frame::BlazingTable>(std::move(csv_table.tbl), csv_table.metadata.column_names);
+		std::unique_ptr<ral::frame::BlazingTable> table_out = std::make_unique<ral::frame::BlazingTable>(std::move(csv_table.tbl), csv_table.metadata.column_names);
+		ral::frame::BlazingTableView table_out_view = table_out->toBlazingTableView();
+		return std::make_pair(std::move(table_out), table_out_view);
```

Comment:
This is incorrect. If the unique_ptr to table is not nullptr then table_out_view is invalid.

Reply:
To be clear either the .first of the pair is nullptr or it is not, If it is not then we dont ever look at table_out_view. Doing it this way is VERY confusing. I am starting to think we should just make BlazingTable be able to be powered from both.

Reply:
Usually what we actually use are BlazingTableViews in the code, BlazingTable is just an owner object. So the function that receives the pair will want to use the BlazingTableView more than the BlazingTable. So if the pattern is that you either use one OR the other, then on the receiving end you would always need to do something like this:
```TableViewPair pair = function()
BlazingTableView my_view = pair.first == nullptr ? pair.second : pair.first.toBlazingTableVIew();```

So we either want every receiving function to handle that, or every function that outputs a pair to make the BlazingTableView for you. I think its better for the function that outputs a pair to handle that.

Also, you dont want the receiving function to be using BLazingTable and the main object it uses, because you dont want to be null checking BlazingTable every single time you want to use it.

Reply:
Yes I am advocating for doing exactly what you describe instead